Neos.Fusion/Classes/Core/Cache/ParserCache.php

```diff
@@ -59,7 +67,18 @@ public function cacheForFusionFile(?string $contextPathAndFilename, \Closure $ge
             $contextPathAndFilename = $this->getAbsolutePathForPackageRessourceUri($contextPathAndFilename);
         }
         $identifier = $this->getCacheIdentifierForFile($contextPathAndFilename);
-        return $this->cacheForIdentifier($identifier, $generateValueToCache);
+        $value = $this->cacheForIdentifier($identifier, $generateValueToCache);
+        if (!$value instanceof FusionFile) {
+            $this->logger->error(sprintf(
```

Comment:
maybe we should even use the throwable error storage to empathise on the error?

Reply:
I wouldn't even categorize this as error. It's a cache, things can happen, it should be able to deal with this.

Reply:
okay now that you say this, i agree.

> Is why I almost never use "has" in caches, it is never atomic, you cannot know what happens between has and get.

will update this accordingly. Is there anything we have to fix in flow then for 9.0? Like remove has :D (No just joking as _false_ could be an acceptable value...

Reply:
removed the logging ;)